internal/impl/global.go

```diff
@@ -143,6 +143,24 @@ func GlobalNixProfilePath() (string, error) {
 	return filepath.Join(path, "profile"), nil
 }
 
+// GenerateShellEnv generates shell commands that configure the user's shell
+// environment to work with Devbox packages. Most notably, it adds Devbox
+// packages to the user's PATH. The commands are intended to be evaluated in
+// the user's shell rcfile. For example:
+//
+//	echo 'eval "$(devbox global shellenv)"' >> ~/.zshrc
+func GenerateShellEnv() string {
+	// If the user has "eval $(devbox global shellenv)" in their shell's
+	// rcfile, then running "devbox shell" will cause these commands to be
+	// evaluated twice (once by the "parent" shell and once by the devbox
+	// shell). Prevent this by making the eval a no-op when we're already
+	// inside a devbox shell.
+	if IsDevboxShellEnabled() {
```

Comment:
This means that if a package is installed while in shell, there's not way for the user to update the current shell to make it work. They would have to quit the shell, add the eval statement to their rc files and start it again.

Maybe we add 

`devbox activate --force`  (or `--refresh`) or something like that? So we can let the user force the configuration while they are in a devbox shell.

Reply:
Another alternative is to have `GenerateShellEnv` set a new env variable `_DEBOX_SHELL_ENV_SET=1` which is independent of shell. This lets apply this configuration only in cases it hasn't been applied. 

We do still run into the issue @ipince brought up that we don't want global packages to supersede local ones. Maybe we splice in the new path instead?